elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -68,12 +70,45 @@ public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec
     public boolean applyPredicateToObject(T object, FilterPredicate filterPredicate, RequestScope requestScope) {
         try {
             String fieldPath = filterPredicate.getFieldPath();
-            com.yahoo.elide.core.RequestScope scope = (com.yahoo.elide.core.RequestScope) requestScope;
+            com.yahoo.elide.core.RequestScope scope = coreScope(requestScope);
             Predicate fn = filterPredicate.getOperator().contextualize(fieldPath, filterPredicate.getValues(), scope);
             return fn.test(object);
         } catch (Exception e) {
             log.error("Failed to apply predicate {}", filterPredicate, e);
             return false;
         }
     }
+
+    /**
+     * Converts FieldExpressionPath value to corresponding list of Predicates
+     *
+     * @param type         entity
+     * @param requestScope request scope
+     * @param method       associated check method name containing FieldExpressionPath
+     * @param defaultPath  path to use if no FieldExpressionPath defined
+     * @return Predicates
+     */
+    protected static Path getFieldPath(Class<?> type, RequestScope requestScope, String method, String defaultPath) {
+        EntityDictionary dictionary = coreScope(requestScope).getDictionary();
+        try {
+            FilterExpressionPath fep = getFilterExpressionPath(type, method, dictionary);
+            return new Path(type, dictionary, fep == null ? defaultPath : fep.value());
+        } catch (NoSuchMethodException | SecurityException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static FilterExpressionPath getFilterExpressionPath(
+            Class<?> type,
+            String method,
+            EntityDictionary dictionary) throws NoSuchMethodException {
+        FilterExpressionPath path = dictionary.lookupEntityClass(type)
+                .getMethod(method)
```

Comment:
This should work for fields as well.

Reply:
The only way to walk a path between beans is with a `@ComputedRelationship`